common/libs/wallet/non-deterministic/web3.ts

```diff
@@ -24,69 +27,53 @@ export default class Web3Wallet implements IWallet {
     );
   }
 
-  public signMessage(msg: string): Promise<string> {
-    return new Promise((resolve, reject) => {
-      const msgHex = bufferToHex(Buffer.from(msg));
-      const options = {
-        method: 'personal_sign',
-        params: [msgHex, this.address],
-        signingAddr: this.address
-      };
+  public async signMessage(msg: string): Promise<string> {
+    const msgHex = bufferToHex(Buffer.from(msg));
+    const state = configuredStore.getState();
+    const nodeLib: Web3Node | INode = getNodeLib(state);
 
-      this.web3.currentProvider.sendAsync(options, (err, data) => {
-        if (err) {
-          return reject(err);
-        }
+    if (!isWeb3Node(nodeLib)) {
+      throw new Error('Web3 wallets can only be used with a Web3 node.');
+    }
 
-        if (data.error) {
-          return reject(data.error);
-        }
-        resolve(data.result);
-      });
-    });
+    return nodeLib.signMessage(msgHex, this.address);
   }
 
-  public sendTransaction(transaction: ExtendedRawTransaction): Promise<string> {
-    return new Promise((resolve, reject) => {
-      const { from, to, value, gasLimit, gasPrice, data, nonce } = transaction;
+  public async sendTransaction(
+    transaction: ExtendedRawTransaction
+  ): Promise<string> {
+    const state = configuredStore.getState();
+    const nodeLib: Web3Node | INode = getNodeLib(state);
+    const { from, to, value, gasLimit, gasPrice, data, nonce } = transaction;
+    const web3Tx = {
+      from,
+      to,
+      value,
+      gas:
+        gasLimit instanceof BN ? sanitizeHex(gasLimit.toString(16)) : gasLimit,
```

Comment:
Would be nice to lock down the gasLimit type to one or the other to put conditionals like this to bed. But definitely not requesting that for this PR.

Reply:
In `transaction-refactor` it only comes in as one type